added basic node server

rhgksrua opened the pull request:
Added basic node server using Express web framework.

running `node server.js` or `npm start` should start the server.  Just remember to run `npm install` before running it.

`resources/` directory contains js and css for client.
`server.js` serves `index.html`.

JordyVlassembrouck commented on karma.conf.js:
```diff
@@ -15,8 +15,8 @@ module.exports = function(config) {
     files: [
-      './js/main/*.js',
-      './js/test/*.js'
+        './public/js/*.js',
```
Following the style guide on the [wiki](https://github.com/DanyDin1801/Product-Portal/wiki), javascript files should be indented with two spaces.

JordyVlassembrouck commented on public/js/index.js:
```diff
@@ -1,3 +1,5 @@
 function displayMessage() {
     return 'Hello, World!';
```
I know this method was written by me, but that was before we had the style guide.
Could you correct the indentation please?

JordyVlassembrouck commented on server.js:
```diff
@@ -0,0 +1,15 @@
+app.use(express.static('public'));
+
+app.get('/', function(req, res) {
+    res.sendFile(__dirname + '/html/index.html');
```
2 spaces indentation rule ;)

JordyVlassembrouck commented on server.js:
```diff
@@ -0,0 +1,15 @@
+});
+
+app.listen(port, function() {
+    console.log('http://localhost:' + port);
```
Same here, 2 spaces.

JordyVlassembrouck commented on test/index_spec.js:
```diff
@@ -0,0 +1,5 @@
+describe('index', function() {
+  it('should return "hello world"', function() {
+      expect(displayMessage()).toBe('Hello, World!');
```
And here. Correct indentation please.

JordyVlassembrouck commented:
@vinaypuppal Good point. But maybe we should do that in a separate issue.
I have seen alot of wrong indentations, so maybe we can create an issue where we implement the nodemon module and a JavaScript linter module, all within the same issue, sounds cool?

rhgksrua commented:
I've fixed all indentation to 2 spaces and added `npm start` as another option to start the server.  I'll leave out `nodemon` for now.  I've read the chat and we'll figure out if we'll use `nodemon` or some kind of live reloading.  @vinaypuppal, can you create an issue for that?

JordyVlassembrouck commented:
You have my approval, @rhgksrua

vinaypuppal commented:
@rhgksrua done created a new issue #6
